thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/ThriftProtocolFactories.java

```diff
@@ -77,59 +83,53 @@ public String toString() {
+    }
+
     /**
      * Returns the {@link TProtocolFactory} for the specified {@link SerializationFormat}.
      *
-     * @throws IllegalArgumentException if the specified {@link SerializationFormat} is not for Thrift
+     * @throws IllegalArgumentException if the specified {@link SerializationFormat} is not a
+     *         known Thrift serialization format
      */
     public static TProtocolFactory get(SerializationFormat serializationFormat) {
         requireNonNull(serializationFormat, "serializationFormat");
-
-        if (serializationFormat == ThriftSerializationFormats.BINARY) {
-            return BINARY;
-        }
-
-        if (serializationFormat == ThriftSerializationFormats.COMPACT) {
-            return COMPACT;
-        }
-
-        if (serializationFormat == ThriftSerializationFormats.JSON) {
-            return JSON;
-        }
-
-        if (serializationFormat == ThriftSerializationFormats.TEXT) {
-            return TEXT;
-        }
-
-        if (serializationFormat == ThriftSerializationFormats.TEXT_NAMED_ENUM) {
-            return TEXT_NAMED_ENUM;
-        }
-
-        throw new IllegalArgumentException("non-Thrift serializationFormat: " + serializationFormat);
+        return Optional.ofNullable(knownProtocolFactories.get(serializationFormat))
+                .orElseThrow(() -> new IllegalArgumentException(
+                        "Unsupported Thrift serializationFormat: " + serializationFormat));
     }
 
     /**
-     * Returns the {@link SerializationFormat} for the specified {@link TProtocolFactory}.
+     * Retrieves all registered Thrift serialization formats.
      *
-     * @throws IllegalArgumentException if the specified {@link TProtocolFactory} is not known by this class
+     * @return an unmodifiable view of the registered Thrift serialization formats set.
      */
-    public static SerializationFormat toSerializationFormat(TProtocolFactory protoFactory) {
-        requireNonNull(protoFactory, "protoFactory");
-
-        if (protoFactory instanceof TBinaryProtocol.Factory) {
-            return ThriftSerializationFormats.BINARY;
-        } else if (protoFactory instanceof TCompactProtocol.Factory) {
-            return ThriftSerializationFormats.COMPACT;
-        } else if (protoFactory instanceof TJSONProtocol.Factory) {
-            return ThriftSerializationFormats.JSON;
-        } else if (protoFactory instanceof TTextProtocolFactory) {
-            final TTextProtocolFactory factory = (TTextProtocolFactory) protoFactory;
-            return factory.usesNamedEnums() ? ThriftSerializationFormats.TEXT_NAMED_ENUM
-                                            : ThriftSerializationFormats.TEXT;
-        } else {
-            throw new IllegalArgumentException(
-                    "unsupported TProtocolFactory: " + protoFactory.getClass().getName());
-        }
+    public static Set<SerializationFormat> getThriftSerializationFormats() {
```

Comment:
How about `values()` that returns `Set<ThriftProtocolFactory>`? `ThriftProtocolFactory` could provide some common properties including `SerializationFormat` and `TProtocolFactory` (or just methods that delegates to `TProtocolFactory`).

Reply:
I don't see any use for it at the moment

Reply:
OK. Then how about making `ThriftSerializationFormats.values()` return what this method returns, and then hiding this method from the public API?